### PR TITLE
Add `Label` class for labeling 3D points

### PR DESCRIPTION
**Pull request by user27182**
### Overview

Follow-up from #6263.

This class is similar to `add_point_labels` but is for a single label instead of many. Its implementation/API is also much simpler. 

Since it inherits from `Text`, it has a `TextProperty` and it therefore supports chinese characters (unlike `add_point_labels`, see https://github.com/pyvista/pyvista/issues/4548#issuecomment-2186799201)

### CODE REVIEW

File: doc/source/api/plotting/index.rst [hunk 1]
```diff
@@ -41,6 +41,7 @@ all plotting functionality in PyVista.
    Renderer
    RenderWindowInteractor
    Text
+   TextLabel
```
Comment (tkoyama010) [inline doc/source/api/plotting/index.rst:44]:
```suggestion
   Label
```

File: pyvista/plotting/__init__.py [hunk 2]
```diff
@@ -54,6 +54,7 @@
 from .renderer import scale_point
 from .text import CornerAnnotation
 from .text import Text
+from .text import TextLabel
```
Comment (tkoyama010) [inline pyvista/plotting/__init__.py:57]:
```suggestion
from .text import Label
```

File: pyvista/plotting/text.py [hunk 3]
```diff
@@ -226,6 +230,85 @@ def position(self, position: Sequence[float]):  # numpydoc ignore=GL08
         self.SetPosition(position[0], position[1])
 
 
+class TextLabel(Text):
```
Comment (tkoyama010) [inline pyvista/plotting/text.py:233]:
```suggestion
class Label(Text):
```

File: pyvista/plotting/text.py [hunk 4]
```diff
@@ -226,6 +230,85 @@ def position(self, position: Sequence[float]):  # numpydoc ignore=GL08
...
+    """2D text actor with a 3D position coordinate.
```
Comment (tkoyama010) [inline pyvista/plotting/text.py:234]:
```suggestion
    """2D label actor with a 3D position coordinate.
```

File: pyvista/plotting/text.py [hunk 5]
```diff
@@ -226,6 +230,85 @@ def position(self, position: Sequence[float]):  # numpydoc ignore=GL08
...
+    pyvista.Plotter.add_point_labels
+
+    Examples
+    --------
+    Create a text label for a point of interest.
+
+    >>> import pyvista as pv
+    >>> mesh = pv.Cone()
+    >>> tip_point = mesh.points[0]
+    >>> label = pv.TextLabel('tip', position=tip_point)
```
Comment (tkoyama010) [inline pyvista/plotting/text.py:261]:
```suggestion
    >>> label = pv.Label('tip', position=tip_point)
```

File: pyvista/plotting/text.py [hunk 6]
```diff
@@ -226,6 +230,85 @@ def position(self, position: Sequence[float]):  # numpydoc ignore=GL08
...
+    prop : pyvista.TextProperty, optional
+        The property of this actor.
+
+    See Also
+    --------
+    pyvista.Plotter.add_point_labels
+
+    Examples
+    --------
+    Create a text label for a point of interest.
```
Comment (tkoyama010) [inline pyvista/plotting/text.py:256]:
```suggestion
    Create a label for a point of interest.
```

File: tests/plotting/test_text.py [hunk 7]
```diff
@@ -48,6 +48,22 @@ def test_text_position(text):
     assert np.all(text.position == position)
 
 
+def test_text_label():
```
Comment (tkoyama010) [inline tests/plotting/test_text.py:51]:
```suggestion
def test_label():
```

File: tests/plotting/test_text.py [hunk 8]
```diff
@@ -48,6 +48,22 @@ def test_text_position(text):
...
+    label = pv.TextLabel('text', (1, 2, 3), size=42, prop=pv.Property())
```
Comment (tkoyama010) [inline tests/plotting/test_text.py:52]:
```suggestion
    label = pv.Label('text', (1, 2, 3), size=42, prop=pv.Property())
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]
Suggestion for class name: Since the Text part of `TextLabel` is redundant, how about changing it to `Label`?
Although unrelated to this, I personally feel that the point in add_point_labels is also redundant.

**Comment by user27182** [comment]
pre-commit.ci autofix

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Pull request merged**